package/cpp/rnskia/dom/base/JsiDomNode.h

```diff
@@ -370,6 +370,11 @@ class JsiDomNode : public JsiHostObject,
     enqueAsynOperation([child, weakSelf = weak_from_this()]() {
       auto self = weakSelf.lock();
       if (self) {
+        // Remove the existing instance of the child, before adding it to the end of the list
```

Comment:
As suspected, this is not needed in addChild but only in insertChildBefore, can you confirm?